### PR TITLE
ref(cli): Add better error message when powershell is missing during upgrade

### PR DESCRIPTION
**Pull request by kamilogorek**
Closes https://github.com/denoland/deno/issues/17756

### CODE REVIEW

File: cli/tools/upgrade.rs [hunk 1]
```diff
@@ -510,7 +510,17 @@ pub fn unpack_into_dir(
         .arg(format!("'{}'", &archive_path.to_str().unwrap()))
         .arg("-DestinationPath")
         .arg(format!("'{}'", &temp_dir_path.to_str().unwrap()))
-        .spawn()?
+        .spawn()
+        .map_err(|err| {
+          if err.kind() == std::io::ErrorKind::NotFound {
+            std::io::Error::new(
+              std::io::ErrorKind::NotFound,
+              "`powershell.exe` was not found on your PATH",
```
Comment (aapoalas) [inline cli/tools/upgrade.rs:518]: **Nitpick:** `in PATH`

**Review by aapoalas — APPROVED** [review]
Nice

File: cli/tools/upgrade.rs (hunk 1 above)
Comment (kamilogorek) [reply cli/tools/upgrade.rs:518]: Copy-pasted from your other error message 🌝

**Comment by aapoalas** [comment]
Thank you!

**Pull request merged**